met-web/src/components/engagement/public/view/EngagementDetailsTabs.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { Suspense, useState, useEffect } from 'react';
+import { Tab, Skeleton, Box, useMediaQuery, Theme } from '@mui/material';
+import { TabContext, TabList, TabPanel } from '@mui/lab';
+import { Await, useLoaderData } from 'react-router-dom';
+import { getEditorStateFromRaw } from 'components/common/RichTextEditor/utils';
+import { Header2 } from 'components/common/Typography';
+import { colors } from 'components/common';
+import { RichTextArea } from 'components/common/Input/RichTextArea';
+import { EngagementLoaderData } from './EngagementLoader';
+import { EngagementViewSections } from '.';
+import { FormDetailsTab } from 'engagements/admin/create/authoring/types';
+
+// Todo: Replace this placeholder widget boolean type with a real widget type
+interface FormDetailsTabWithWidget extends FormDetailsTab {
+    widget: boolean;
+}
+
+export const EngagementDetailsTabs = () => {
+    const { details } = useLoaderData() as EngagementLoaderData;
+    const [detailsTabs, setDetailsTabs] = useState<FormDetailsTabWithWidget[]>([]);
+    const [selectedTab, setSelectedTab] = useState('0');
+    const isMobile = useMediaQuery((theme: Theme) => theme.breakpoints.down('sm'));
+
+    useEffect(() => {
+        details.then((tabs) => {
+            const parsedTabs: FormDetailsTabWithWidget[] = tabs.map((t) => ({
+                id: t.id || -1,
+                engagement_id: t.engagement_id || 0,
+                label: t.label || '',
+                slug: t.slug || '',
+                heading: t.heading || '',
+                body: getEditorStateFromRaw(JSON.stringify(t.body) || ''),
+                sort_index: t.sort_index || -1,
+                widget: true, // Todo: Replace with real widget value
+            }));
+            const sortedTabs = [...parsedTabs].sort((a, b) => a.sort_index - b.sort_index);
+            setDetailsTabs(sortedTabs);
+        });
+    }, [details]);
```

Comment:
With react-router we should very rarely have to rely on `useEffect`s for state management. You can apply your transforms to the promise with `.then()`, then await that where you render your TabPanels.

```suggestion
    const sortedDetailsTabs = details.then((tabs) => {
        const parsedTabs: FormDetailsTabWithWidget[] = tabs.map((t) => ({
            id: t.id || -1,
            engagement_id: t.engagement_id || 0,
            label: t.label || '',
            slug: t.slug || '',
            heading: t.heading || '',
            body: getEditorStateFromRaw(JSON.stringify(t.body) || ''),
            sort_index: t.sort_index || -1,
            widget: true, // Todo: Replace with real widget value
        }));
        const sortedTabs = [...parsedTabs].sort((a, b) => a.sort_index - b.sort_index);
        return sortedTabs;
    })
```